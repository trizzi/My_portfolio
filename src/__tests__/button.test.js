import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { useTheme } from '../context/ThemeContext';
import Button from '../components/button/Button';

// Mock the ThemeContext to provide a mock theme
jest.mock('../context/ThemeContext.js', () => ({
	useTheme: jest.fn(),
}));

describe('Button component', () => {
	// Mock theme values
	const mockTheme = {
		primary: {
			purple: '#6c63ff',
		},
		transparent: 'transparent',
		textcolor: {
			light: '#ffffff',
			black: '#000000',
		},
	};

	beforeEach(() => {
		// Mock the useTheme hook to return the mock theme
		useTheme.mockReturnValue({ theme: mockTheme });
	});

	test('renders button with correct styles', () => {
		render(
			<Button
				text='Click me'
				background
				size='lg'
				disable={false}
				time={2}
			/>,
		);

		// Check if the button is rendered with correct styles
		const button = screen.getByText('Click me');
		expect(button).toBeInTheDocument();
		expect(button).toHaveStyle(`
      background-color: ${mockTheme.primary.purple};
      color: ${mockTheme.textcolor.light};
      border: 2px solid ${mockTheme.primary.purple};
    `);
	});

	test('Button component matches snapshot', () => {
		const { container } = render(<Button text='Click me' />);
		expect(container).toMatchSnapshot();
	});
});
