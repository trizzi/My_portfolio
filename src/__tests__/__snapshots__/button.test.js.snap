// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button component matches snapshot 1`] = `
<div>
  <button
    class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
    style="background-color: transparent; color: rgb(0, 0, 0); border: 2px solid #6c63ff; opacity: 0;"
    tabindex="0"
  >
    Click me
  </button>
</div>
`;
